src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -496,267 +550,202 @@ using namespace std;
+                for (size_t i = 0; i < param->get_output_size(); ++i)
                 {
+                    shared_ptr<descriptor::TensorView> tv = param->get_output_tensor_view(i);
+                    const element::Type& et = tv->get_tensor_view_type()->get_element_type();
+                    string type = et.c_type_string();
                     stringstream ss;
-                    ss << "((" << tensor->get_element_type().c_type_string()
-                       << "*)((char *)pool_base_ptr + " << tensor->get_pool_offset() << "))";
-                    m_variable_name_map[tensor->get_name()] = ss.str();
+                    ss << "((" << type << "*)(inputs[" << arg_index << "]))";
+                    m_variable_name_map[tv->get_tensor().get_name()] = ss.str();
+                    arg_index++;
                 }
             }
-        }
 
-        // Add inputs to the variable name map
-        size_t arg_index = 0;
-        for (shared_ptr<ngraph::op::Parameter> param : current_function->get_parameters())
-        {
-            for (size_t i = 0; i < param->get_output_size(); ++i)
+            // Add outputs to the variable name map
+            for (size_t i = 0; i < current_function->get_output_size(); ++i)
             {
-                shared_ptr<descriptor::TensorView> tv = param->get_output_tensor_view(i);
-                const element::Type& et = tv->get_tensor_view_type()->get_element_type();
-                string type = et.c_type_string();
+                shared_ptr<Node> op = current_function->get_output_op(i);
+                shared_ptr<descriptor::TensorView> tv = op->get_output_tensor_view();
+                string type = tv->get_tensor_view_type()->get_element_type().c_type_string();
                 stringstream ss;
-                ss << "((" << type << "*)(inputs[" << arg_index << "]))";
+                ss << "((" << type << "*)(outputs[" << i << "]))";
                 m_variable_name_map[tv->get_tensor().get_name()] = ss.str();
-                arg_index++;
-            }
-        }
 
-        // create output alias map
-        size_t output_index = 0;
-        unordered_map<descriptor::TensorView*, vector<size_t>> output_alias_map;
-        vector<size_t> aliases;
-        for (size_t i = 0; i < current_function->get_output_size(); ++i)
-        {
-            shared_ptr<Node> op = current_function->get_output_op(i);
-            shared_ptr<descriptor::TensorView> otv = op->get_output_tensor_view();
-            vector<size_t>& al = output_alias_map[otv.get()];
-            al.push_back(output_index);
-            if (al.size() > 1)
-            {
-                aliases.push_back(output_index);
+                //it should be safe to assign both descriptors to one output*
+                //since needs_copy == false makes `op::Result` an nop
+                auto res = dynamic_pointer_cast<ngraph::op::Result>(op);
+                if (!res->needs_copy())
```

Comment:
Nice, it looks like you have added support for `ngraph::pass::ResultCopyElimination`. Have you tried adding `    pass_manager.register_pass<ngraph::pass::ResultCopyElimination>();` to see if it functions as expected?

Reply:
let me try it.

Reply:
Looks like it works.

Reply:
added.